mlos_bench/mlos_bench/services/remote/azure/azure_flex.py

```diff
@@ -0,0 +1,207 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+A collection Service functions for managing VMs on Azure.
+"""
+import logging
+
+from typing import Any, Dict, Optional, Tuple
+
+import requests
+
+from mlos_bench.environments.status import Status
+from mlos_bench.services.base_service import Service
+from mlos_bench.services.types.authenticator_type import SupportsAuth
+from mlos_bench.services.types.remote_config_type import SupportsRemoteConfig
+from mlos_bench.util import check_required_params, merge_parameters
+
+_LOG = logging.getLogger(__name__)
+
+
+class AzureFlexConfigService(Service, SupportsRemoteConfig):
+    """
+    Helper methods to configure Azure Flex services.
+    """
+
+    _REQUEST_TIMEOUT = 5  # seconds
+
+    # Azure Flex Services Configuration REST API as described in
+    # https://learn.microsoft.com/en-us/rest/api/mysql/flexibleserver/configurations
+
+    _URL_CONFIGURE = (
+        "https://management.azure.com" +
+        "/subscriptions/{subscription}" +
+        "/resourceGroups/{resource_group}" +
+        "/providers/{provider}" +
+        "/{server_type}/{vm_name}" +
+        "/{update}" +
+        "?api-version={api_version}"
+    )
+
+    def __init__(self,
+                 config: Optional[Dict[str, Any]] = None,
+                 global_config: Optional[Dict[str, Any]] = None,
+                 parent: Optional[Service] = None):
+        """
+        Create a new instance of Azure services proxy.
+
+        Parameters
+        ----------
+        config : dict
+            Free-format dictionary that contains the benchmark environment
+            configuration.
+        global_config : dict
+            Free-format dictionary of global parameters.
+        parent : Service
+            Parent service that can provide mixin functions.
+        """
+        super().__init__(config, global_config, parent)
+
+        check_required_params(self.config, {
+            "subscription",
+            "resourceGroup",
+            "provider",
+        })
+
+        (self._is_batch, is_flex, api_version) = {
+            "Microsoft.DBforMySQL": (True, True, "2022-01-01"),
+            "Microsoft.DBforMariaDB": (False, False, "2018-06-01"),
+            "Microsoft.DBforPostgreSQL": (False, True, "2022-12-01"),
+        }[self.config["provider"]]
```

Comment:
This makes me a little nervous - seems like a support headache.  Is there a more generic way we can do more of this with just configs?